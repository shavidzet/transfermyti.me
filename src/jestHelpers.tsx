/* eslint-disable import/no-extraneous-dependencies,@typescript-eslint/no-explicit-any,no-console */
import React from "react";
import {
  RenderOptions,
  RenderResult,
  render as testRender,
} from "@testing-library/react";
import { ThemeProvider } from "emotion-theming";
import { theme } from "./components/emotion";

// These are exported for convenience (so we don't have to add an additional import
// to the test file).
export { createEvent, fireEvent, RenderResult } from "@testing-library/react";
export { theme } from "./components/emotion";

/**
 * Renders the specified component with react-testing-library wrapped in a theme
 * provider so we don't have to do it explicitly for each component.
 */
export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult {
  return testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options);
}

type LogMethod = "log" | "warn" | "error";

export class ConsoleToggle {
  private consoleMethod: (...args: any[]) => void = console.warn;
  private readonly logMethod: LogMethod;

  public constructor(logMethod: LogMethod) {
    this.logMethod = logMethod;
  }

  public onBefore(): void {
    this.consoleMethod = console[this.logMethod];
    console[this.logMethod] = jest.fn();
  }

  public onAfter(): void {
    console[this.logMethod] = this.consoleMethod;
  }
}
