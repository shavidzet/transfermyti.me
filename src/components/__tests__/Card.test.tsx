import React from "react";
import { render, RenderResult } from "~/jestHelpers";
import Card from "../Card";

const setup = (
  propOverrides: any = {},
): { props: any; wrapper: RenderResult } => {
  const props = {
    title: "Test Title",
    ...propOverrides,
  };

  const wrapper = render(<Card {...props} />);

  return { props, wrapper };
};

describe("the <Card> component", () => {
  test("matches its snapshot with valid props", () => {
    const { wrapper } = setup();

    expect(wrapper.baseElement.firstElementChild).toMatchSnapshot();
  });
});
