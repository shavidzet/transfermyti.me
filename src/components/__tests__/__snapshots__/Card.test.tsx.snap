// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <Card> component matches its snapshot with valid props 1`] = `
.emotion-0 {
  border-radius: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 20rem;
  -ms-flex: 0 0 20rem;
  flex: 0 0 20rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0 1rem 2rem;
  padding: 1rem;
  background: rgb(30,120,161);
  box-shadow: 0 2px 4px -1px rgba(0,0,0,0.2), 0 4px 5px 0 rgba(0,0,0,0.14), 0 1px 10px 0 rgba(0,0,0,0.12);
  color: rgb(239,253,232);
}

@media (max-width:32rem) {
  .emotion-0 {
    -webkit-flex-basis: 16rem;
    -ms-flex-preferred-size: 16rem;
    flex-basis: 16rem;
  }
}

<div>
  <li
    class="emotion-0"
  >
    <h2>
      Test Title
    </h2>
  </li>
</div>
`;
